### PR TITLE
Feat: add hooks for NFTs

### PR DESCRIPTION
**Pull request by EsterB12**
added hooks for public NFT sales
added hooks for private NFT sales

### CODE REVIEW

File: hooks/useCreateSale.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+      status: "error",
+      isClosable: true
+    });
+    return;
+  }
+
+  const createSaleBatch = async (
+    tokenIds: number[],
+    saleStartTime: number,
+    floorPrice: number,
```
Comment (saksijas) [inline hooks/useCreateSale.tsx:24]: parseUnits is needed here

File: services/web3/useCreatePsycSale.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    ) => {
+      return writeContract({
+        address: psycSaleSepolia,
+        functionName: "createSaleBatch",
+        abi: psycSaleAbiSepolia,
+        args: [
+          tokenIds,
+          saleStartTime,
+          floorPrice,
+          ceilingPrice,
```
Comment (saksijas) [inline services/web3/useCreatePsycSale.tsx:31]: check if we need to send bigInt with eth decimals

File: services/web3/useBuySpecificPsycPrivate.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+      tokenId: number,
+      proof: string
+    ) => {
+      const nftAmount = parseUnits(buyFromBatch, 18);
+      return writeContract({
+        address: psycSaleSepolia,
+        functionName: "buyFromBatch",
+        abi: psycSaleAbiSepolia,
+        args: [batchId, tokenId, proof],
+        value: nftAmount
```
[the remark this answers is not in the document]
Comment (EsterB12) [reply services/web3/useBuySpecificPsycPrivate.tsx:28]: parseUnits gets called on line 22!

File: services/web3/useBuySpecificPsycCopy.tsx [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+      batchId: number,
+      erc721TokenId: number
+    ) => {
+      const nftCopyAmount = parseUnits(buyNftCopyFromBatch, 18);
+      return writeContract({
+        address: psycSaleSepolia,
+        functionName: "buyNftCopyFromBatch",
+        abi: psycSaleAbiSepolia,
+        args: [batchId, erc721TokenId],
+        value: nftCopyAmount
```
[the remark this answers is not in the document]
Comment (EsterB12) [reply services/web3/useBuySpecificPsycCopy.tsx:27]: parseUnits gets called on line 21!

File: services/web3/useBuyRandomPsycPublic.tsx [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+  const buyRandomPsycPublic = useCallback(
+    async (buyRandomFromBatch: string, batchId: number) => {
+      const randomNftCopyAmount = parseUnits(buyRandomFromBatch, 18);
+      return writeContract({
+        address: psycSaleSepolia,
+        functionName: "buyRandomNftCopyFromBatch",
+        abi: psycSaleAbiSepolia,
+        args: [batchId],
+        value: randomNftCopyAmount
```
[the remark this answers is not in the document]
Comment (EsterB12) [reply services/web3/useBuyRandomPsycPublic.tsx:22]: parseUnits gets called on line 16!

File: services/web3/useBuyRandomPsycPrivate.tsx [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+  const buyRandomPsycPrivate = useCallback(
+    async (buyRandomFromBatch: string, batchId: number, proof: string) => {
+      const randomNftAmount = parseUnits(buyRandomFromBatch, 18);
+      return writeContract({
+        address: psycSaleSepolia,
+        functionName: "buyRandomFromBatch",
+        abi: psycSaleAbiSepolia,
+        args: [batchId, proof],
+        value: randomNftAmount
```
[the remark this answers is not in the document]
Comment (EsterB12) [reply services/web3/useBuyRandomPsycPrivate.tsx:22]: parseUnits gets called on line 16 and wraps to the next!

File: services/web3/useCreatePsycSale.tsx (hunk 2 above)
Comment (EsterB12) [reply services/web3/useCreatePsycSale.tsx:31]: I will! I'll resolve this today

Comment (EsterB12) [reply services/web3/useCreatePsycSale.tsx]: it needed to be bigint
@saksijas I made those changes a while ago

**Pull request merged**